diofant/polys/rootoftools.py

```diff
@@ -299,9 +303,52 @@ def _complexes_sorted(cls, complexes):
             return []
         cache = {}
 
+        def is_re_disjoint(u, v):
+            if u.conj != v.conj:
+                return True
+            else:
+                return u.bx <= v.ax or v.bx <= u.ax
+
+        res_cache = {}
+
+        def match_same_real_part(u, v):
+            assert u.domain == v.domain
+            dom = u.domain
+            if tuple(u.f) not in res_cache:
+                res_cache[tuple(u.f)] = dmp_resultant(dmp_permute(u.f1, [1, 0], 1, dom),
```

Comment:
Where are `u.f`, `u.f1`, `u.f2` defined?

Reply:
@jksuom, they are used (or not, as for ``f``) - in the ComplexInterval class for bookeeping.  ``f`` - original polynomial (AFAIK, not saved in Sympy).  ``f1`` and ``f2`` - its real and imaginary parts.